File: web_admin/lib/data/mappers/manga_mapper.dart
import 'package:web_admin/data/models/manga.dart';
import 'package:web_admin/domain/entities/manga.dart';

extension MangaModelMapper on MangaModel {
  MangaEntity toEntity() {
    return MangaEntity(
      id: id,
      title: title,
      description: description,
      thumbnail: thumbnail,
      status: status,
      totalChapter: totalChapter,
      rate: rate,
      authorId: authorId,
      genreIds: genreIds,
      releaseDate: releaseDate,
      endDate: endDate,
    );
  }
}

extension MangaEntityMapper on MangaEntity {
  MangaModel toModel() {
    return MangaModel(
      id: id,
      title: title,
      description: description,
      thumbnail: thumbnail,
      status: status,
      totalChapter: totalChapter,
      rate: rate,
      authorId: authorId,
      genreIds: genreIds,
      releaseDate: releaseDate,
      endDate: endDate,
    );
  }
}

extension MangaModelListMapper on List<MangaModel> {
  List<MangaEntity> toEntityList() {
    return map((mangaModel) => mangaModel.toEntity()).toList();
  }
}

extension MangaEntityListMapper on List<MangaEntity> {
  List<MangaModel> toModelList() {
    return map((mangaEntity) => mangaEntity.toModel()).toList();
  }
}

File: backend/src/Mappers/MangaMapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using backend.src.Dtos.Manga;
using backend.src.Models;

namespace backend.src.Mappers
{
    public static class MangaMapper
    {
        public static async Task<Manga> FromDtoToManga(this CreateMangaDto dto)
        {
            return new Manga
            {
                Id = dto.Id,
                Title = dto.Title,
                Description = dto.Description,
                Thumbnail = dto.Thumbnail,
                Status = dto.Status,
                TotalChapter = dto.TotalChapter,
                Rate = dto.Rate,
                ReleaseDate = dto.ReleaseDate,
                EndDate = dto.EndDate,
                AuthorId = dto.AuthorId,
                GenreIds = dto.GenreIds ?? new List<int>()
            };
        }
    }
}